### PR TITLE
properties for poi object

### PR DESCRIPTION
**Pull request by azime**

### CODE REVIEW

File: libs/bragi/src/model.rs [hunk 1]
```diff
@@ -78,8 +79,11 @@ pub struct GeocodingResponse {
     pub administrative_regions: Vec<Rc<mimir::Admin>>,
     #[serde(skip_serializing_if = "Vec::is_empty", default)]
     pub poi_types: Vec<mimir::PoiType>,
+    #[serde(skip_serializing_if = "BTreeMap::is_empty")]
```
Comment (pbougue) [inline libs/bragi/src/model.rs]: (@TeXitoi) might more secure with a `default` here too, no? https://serde.rs/field-attrs.html#serdedefault

File: src/osm_reader/poi.rs [hunk 2]
```diff
@@ -191,6 +191,13 @@ const DEFAULT_JSON_POI_TYPES: &'static str = r#"
 }
 "#;
 
+fn make_properties(tags: &osmpbfreader::Tags) -> BTreeMap<String, String>{
+    let mut result = BTreeMap::new();
+    for tag in tags.iter(){
+        result.insert(tag.0.clone(), tag.1.clone());
+    }
+    result
+}
```
Comment (pbougue) [inline src/osm_reader/poi.rs]: fmt: probably missing newline

Comment (antoine-de) [reply src/osm_reader/poi.rs]: isn't there a more rustic way to do this ? it seems a simple conversion between a flat_map and a BTreeMap that could be done with `into_iter` no ?

File: tests/bragi_poi_test.rs [hunk 3]
```diff
@@ -89,6 +89,8 @@ fn poi_admin_address_test(bragi: &BragiHandler) {
 
     // the first element returned should be the poi 'Le-Mée-sur-Seine Courtilleraies'
     let poi = geocodings.first().unwrap();
+	let properties = poi.get("properties").and_then(|json| json.as_array()).unwrap();
+	assert_eq!(properties.len(), 9);
```
Comment (pbougue) [inline tests/bragi_poi_test.rs]: It could be nice to have a test on key/value fields

File: libs/mimir/src/objects.rs [hunk 4]
```diff
@@ -145,7 +145,11 @@ impl<T: MimirObject> MimirObject for Rc<T> {
         T::es_id(self)
     }
 }
-
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Propertie {
```
Comment (pbougue) [inline libs/mimir/src/objects.rs]: typo: Property

File: src/osm_reader/poi.rs [hunk 5]
```diff
@@ -191,6 +191,13 @@ const DEFAULT_JSON_POI_TYPES: &'static str = r#"
 }
 "#;
 
+fn make_properties(tags: &osmpbfreader::Tags) -> Vec<mimir::Propertie>{
+	tags.iter().map(|propretie| mimir::Propertie {
```
Comment (pbougue) [inline src/osm_reader/poi.rs]: typo: property, not propretie

**Comment by pbougue** [comment]
And for rustfmt, we can do something like what's mentionned on that reddit : https://www.reddit.com/r/rust/comments/7l26y1/compliment_for_rustfmt_contributors/drj500o/
The first example looks simple enough (even if we would use it a little bit differently) : https://github.com/killercup/cargo-edit/blob/7a3964c08f363e4865fca531660414c2627ef901/.travis.yml#L9-L15

File: src/osm_reader/poi.rs [hunk 6]
```diff
@@ -191,6 +191,13 @@ const DEFAULT_JSON_POI_TYPES: &'static str = r#"
 }
 "#;
 
+fn make_properties(tags: &osmpbfreader::Tags) -> Vec<mimir::Property>{
+	tags.iter().map(|propretie| mimir::Property {
```
Comment (pbougue) [inline src/osm_reader/poi.rs]: typo: propretie > property

**Pull request merged**